### ISSUE TITLE
How to calculate prompt tokens for embedding model encode?

### ISSUE THREAD

**Issue by gaoxt1983** [UKPLab/sentence-transformers#3416]
I want to calculate input prompt tokens, which returns to user to let them know how many tokens they consumed. How can I do that? Could you give me an example?

**Comment by tomaarsen** [comment]
Hello!

Definitely. You can update the `output_value` option when calling `model.encode`/`model.encode_query`/`model.encode_document` to get all of the underlying data used in processing. Then you can count the `input_ids` or `attention_mask` to get the number of processed tokens, either with or without padding tokens:
```python
import torch
from sentence_transformers import SentenceTransformer

model = SentenceTransformer("Qwen/Qwen3-Embedding-0.6B")

texts = [
    "The quick brown fox jumps over the lazy dog.",
    "A journey of a thousand miles begins with a single step.",
    "To be or not to be, that is the question.",
    "All that glitters is not gold.",
]
# By setting output_value=None, we get all behind-the-scenes features, not just the output embeddings
outputs = model.encode(texts, output_value=None)
# We can find the embeddings with the key 'sentence_embedding'
embeddings = torch.stack([output['sentence_embedding'] for output in outputs])
print(embeddings.shape)
# torch.Size([4, 1024])

# And the input IDs with the key 'input_ids'
input_ids = [output['input_ids'] for output in outputs]

# If you want your token count to include padding tokens, you can sum the lengths of the input IDs
token_count = sum(len(ids) for ids in input_ids)
print(token_count)
# 52

# Alternatively, if you want to count only the non-padding tokens, you can use the 'attention_mask'
attention_masks = [output['attention_mask'] for output in outputs]
non_padding_token_count = sum(mask.sum().item() for mask in attention_masks)
print(non_padding_token_count)
# 46
```

- Tom Aarsen
